### PR TITLE
Add pass-through info to OpSchema to add shared data to stage outputs.

### PR DESCRIPTION
**Pull request by mzient**
Signed-off-by: Michal Zientkiewicz <michalz@nvidia.com>

#### Why we need this PR?
*Pick one, remove the rest*
- It fixes a bug *bug description*
- It adds new feature needed because of *why we need this feature*
- Refactoring to improve *what*

#### What happened in this PR?
*Fill relevant points, put NA otherwise. Replace anything inside []*
 - What solution was applied:
     * Added pass-through info to OpSchema (mapping from inputs to outputs)
     * Track pass-through tensors in graph
 - Affected modules and functionalities:
     * OpGraph, Executor
 - Key points relevant for the review:
     * Allocation of multiple buffers
 - Validation and testing:
     * ReshapeTest sort-of should do
 - Documentation (including examples):
     * N/A

**JIRA TASK**: N/A

### CODE REVIEW

**Comment by mzient** [comment]
!build

File: dali/test/python/test_operator_reshape.py [hunk 1]
```diff
@@ -53,7 +53,7 @@ def define_graph(self):
           images = images.gpu()
         reshaped = self.reshape(images)
 
-        return [images, reshaped]
+        return [images+0, reshaped]
```
Comment (JanuszL) [inline dali/test/python/test_operator_reshape.py]: Some comment what this trick does, please.

File: dali/pipeline/operator/op_schema.h [hunk 2]
```diff
@@ -456,6 +456,15 @@ class DLL_PUBLIC OpSchema {
     return *this;
   }
 
+  /**
+   * @brief Informs that the data passes though this operator unchanged, only
+   *        the metadata is affected.
+   */
```
Comment (JanuszL) [inline dali/pipeline/operator/op_schema.h]: Please add some info about what should be inside the `inout`.
How this going to work with operators that may have a different amount of in/outs depending on arguments.

File: dali/pipeline/graph/graph_descr.cc [hunk 3]
```diff
@@ -514,26 +512,67 @@ void OpGraph::GenerateDOTFromGraph(const TensorNode &current_node, std::ofstream
...
+bool OpGraph::HasConsumersInOtherStage(const TensorNode &tensor, OpType this_stage) const {
+  for (auto cons_edge : tensor.consumers) {
+    // We found a consumer from different stage, this tensor is a stage output
+    const OpNode &cons_op = Node(cons_edge.node);
+    if (cons_op.op_type != this_stage) {
+      return true;
+    }
+    const OpSchema &schema = cons_op.spec.GetSchema();
+    int out_idx = schema.GetPassThroughOutputIdx(cons_edge.index);
+    while (out_idx >= 0) {
```
Comment (JanuszL) [inline dali/pipeline/graph/graph_descr.cc]:
```suggestion
    if (out_idx >= 0) {
```

**Comment by mzient** [comment]
!build

**Pull request merged**